custom/icds_reports/views.py

```diff
@@ -2060,25 +2062,33 @@ def get(self, request, *args, **kwargs):
 class CasDataExport(View):
     def post(self, request, *args, **kwargs):
         data_type = request.POST.get('indicator', None)
-        state_id = request.POST.get('location', None)
+        location_id = request.POST.get('location', None)
         month = int(request.POST.get('month', None))
         year = int(request.POST.get('year', None))
         selected_date = date(year, month, 1).strftime('%Y-%m-%d')
 
-        if state_id and not user_can_access_location_id(
-                self.kwargs['domain'], request.couch_user, state_id
+        if location_id and not user_can_access_location_id(
+                self.kwargs['domain'], request.couch_user, location_id
         ):
             return JsonResponse({"message": "Sorry, you do not have access to that location."})
 
-        sync, _ = get_cas_data_blob_file(data_type, state_id, selected_date)
+        location = SQLLocation.objects.get(location_id=location_id)
+        state = location
+        while not state.location_type.name == 'state':
+            state = state.parent
+        sync, _ = get_cas_data_blob_file(data_type, state.location_id, selected_date)
```

Comment:
Using `_` as a dummy name is risky, especially in a views file, given how often we use it as an alias for `gettext`.